### PR TITLE
Support consistent block number in rosetta

### LINKED ISSUE #3819: Rosetta support consistent block number
### Problem

Historically, due to lack of consistent block number, mirror node rosetta implementation sets the block whose consensus end timestamp is at or after the first ingested account balance file as block 0.  

Mirror node importer now supports consistent block numbers, i.e., regardless of the startDate the importer is configured with, a block generated at a specific consensus timestamp will always have the same block number.

We need to adjust the logic in rosetta server to present consistent block numbers;

### Solution

replace the current block number offset algorithm with consistent block numbers written by importer. Note we may need to add a configurable option for the genesis block number for mainnet and testnet, since the first block after the first account balance file for the networks may not be block 0.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds consistent block number support in rosetta

- Use record file index as block index in rosetta api
- Update rosetta github workflow to use genesis block index + 1 for start index of data validation
- Update rosetta postman tests to query genesis block index dynamically
- Update charts workflow and rosetta helm chart to use postman test suite from the PR branch

**Related issue(s)**:

Fixes #3819

**Notes for reviewer**:

After we implement data retention, when old data is removed, rosetta server will report block not found error when handling requests for those blocks, however the genesis block index will not change. Once rosetta server restarts, the genesis block index will change to the index of the first record file with consensus end > first account balance file's timestamp.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: charts/hedera-mirror-rosetta/templates/tests/pod.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
       imagePullPolicy: {{ .Values.test.image.pullPolicy }}
       args:
         - run
-        - https://raw.githubusercontent.com/hashgraph/hedera-mirror-node/{{ regexReplaceAll "(\\d+\\.\\d+\\.\\d+(-\\w+)?)" .Chart.AppVersion "v${1}" }}/hedera-mirror-rosetta/scripts/validation/postman/rosetta-api-postman.json
+        - https://raw.githubusercontent.com/hashgraph/hedera-mirror-node/{{ .Values.test.githubRef }}/hedera-mirror-rosetta/scripts/validation/postman/rosetta-api-postman.json
```
Comment (steven-sheehy) [inline charts/hedera-mirror-rosetta/templates/tests/pod.yaml:18]: This syntax won't work without templating. Instead revert and do something like

`regexReplaceAll "(\\d+\\.\\d+\\.\\d+(-\\w+)?)" (default .Values.test.githubRef  .Chart.AppVersion "v${1}"`

File: charts/hedera-mirror-rosetta/values.yaml [hunk 2]
```diff
@@ -223,6 +223,7 @@ terminationGracePeriodSeconds: 60
 
 test:
   enabled: true
+  githubRef:
```
Comment (steven-sheehy) [inline charts/hedera-mirror-rosetta/values.yaml:226]:
```suggestion
  githubRef: ""
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: charts/hedera-mirror-rosetta/templates/tests/pod.yaml (hunk 1 above)
Comment (xin-hedera) [reply charts/hedera-mirror-rosetta/templates/tests/pod.yaml]: fixed

File: .github/workflows/charts.yml [hunk 3]
```diff
@@ -53,4 +53,10 @@ jobs:
       - name: Install chart
         run: |
           yq -i '(.global.image.registry="registry:5000") | (.global.image.tag="${{ env.IMAGE_TAG }}")' charts/hedera-mirror/ci/*.yaml
-          ct install --config .github/ct.yaml --charts=charts/hedera-mirror
+          if [ -n "$GITHUB_HEAD_REF" ]; then
+            # GITHUB_HEAD_REF is only set when the triggering event is either pull_request or pull_request_target. If so
+            # pass an extra helm set arg to provide the correct github ref
+            GIT_REF=${$GITHUB_HEAD_REF#refs/*/}
+            HELM_EXTRA_SET_ARGS="--helm-extra-set-args=\"--set=rosetta.test.githubRef=${GIT_REF}\""
+          fi
```
[the remark this answers is not in the document]
Comment (xin-hedera) [reply .github/workflows/charts.yml]: moved to a separate step

File: .github/workflows/charts.yml [hunk 4]
```diff
@@ -49,8 +49,21 @@ jobs:
...
       - name: Install chart
         run: |
-          yq -i '(.global.image.registry="registry:5000") | (.global.image.tag="${{ env.IMAGE_TAG }}")' charts/hedera-mirror/ci/*.yaml
-          ct install --config .github/ct.yaml --charts=charts/hedera-mirror
+          HELM_EXTRA_SET_ARGS="--set=global.image.registry=registry:5000 --set=global.image.tag=${IMAGE_TAG}"
+          if [ -n "$GIT_REF" ]; then
+               HELM_EXTRA_SET_ARGS="${HELM_EXTRA_SET_ARGS} --set=rosetta.test.githubRef=${GIT_REF}"
+          fi
+          ct install --config .github/ct.yaml --charts=charts/hedera-mirror \
+            --helm-extra-set-args="${HELM_EXTRA_SET_ARGS}"
```
Comment (steven-sheehy) [inline .github/workflows/charts.yml:69]:
```suggestion
        run: ct install --config .github/ct.yaml --charts=charts/hedera-mirror --helm-extra-set-args="--set=global.image.registry=registry:5000 --set=global.image.tag=${IMAGE_TAG} --set=rosetta.test.githubRef=${GIT_REF}"
```

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**